core/app/models/spree/order_cancellations.rb

```diff
@@ -118,7 +118,9 @@ def update_shipped_shipments(inventory_units)
       to_a
 
     shipments.each do |shipment|
-      if shipment.inventory_units.all? { |iu| iu.shipped? || iu.canceled? }
+      if shipment.inventory_units_canceled?
```

Comment:
@kennyadsl I likely need to add a check here if the shipment is not in a shipped state.

Reply:
I'm having a look at the state machine of the shipment and it seems like it's not allowed to move a shipment from `shipped` to `canceled` already: 
 https://github.com/solidusio/solidus/blob/df68c0c748ded6a595d28c37a9460126c6d12ecb/core/lib/spree/core/state_machines/shipment.rb#L36

Do you know if there's a specific reason not to rely on the state machine here? Same for when we set the state to `shipped`. I think we are also skipping the rest of the callbacks defined in the state machine.

Reply:
No, I should be able to use the state machine call. I was mimicking the original code and thought that there must have been a reason to jump the state machine. Although, we should likely use it.

For the code on line 124, should we update that code as well to utilize the state machine?

Reply:
Yes, I think the best path forward at this point is to make another PR that just changes `OrderCancelations` to try to use the state machine. If that works and we can't find blockers, changing this PR will be smooth. What do you think?